File: yapcore/src/main/java/co/yap/yapcore/helpers/validation/rule/DateRule.java
package co.yap.yapcore.helpers.validation.rule;

import android.widget.TextView;

import co.yap.yapcore.helpers.validation.util.DateValidator;
import co.yap.yapcore.helpers.validation.util.EditTextHandler;


/**
 * Created irfan arshad on 10/6/2020.
 */
public class DateRule extends Rule<TextView, String> {

    private final DateValidator dateValidator;

    public DateRule(TextView view, String value, String errorMessage) {
        super(view, value, errorMessage);
        dateValidator = new DateValidator();
    }

    @Override
    public boolean isValid(TextView view) {
        return dateValidator.isValid(view.getText().toString(), value);
    }

    @Override
    public void onValidationSucceeded(TextView view) {
        EditTextHandler.removeError(view);
    }

    @Override
    public void onValidationFailed(TextView view) {
        EditTextHandler.setError(view, errorMessage);
    }
}

File: yapcore/src/main/java/co/yap/yapcore/helpers/validation/rule/DateRule.kt
package co.yap.yapcore.helpers.validation.rule

import android.widget.TextView
import androidx.annotation.Keep
import co.yap.yapcore.helpers.validation.util.DateValidator
import co.yap.yapcore.helpers.validation.util.EditTextHandler.removeError

/**
 * Created irfan arshad on 10/6/2020.
 */
@Keep
class DateRule(
    view: TextView?,
    value: String?,
    errorMessage: String?,
    showErrorMessage: Boolean
) : Rule<TextView?, String?>(
    view,
    value,
    errorMessage,
    showErrorMessage
) {
    private val dateValidator: DateValidator = DateValidator()
    override fun isValid(view: TextView?): Boolean {
        return dateValidator.isValid(view?.text.toString(), value)
    }

    override fun onValidationSucceeded(view: TextView?) {
        removeError(view)
    }

    override fun onValidationFailed(view: TextView?) {
//        setError(view, errorMessage)
    }

}